online-shop/src/reducers/combine.tsx

```diff
@@ -0,0 +1,21 @@
+import { combineReducers } from "redux";
+import { productManipulation } from "./product";
+import productListReducer, { ProductListState } from "./productList";
+import { IStateDetails, IStateCart, IStateEdit } from "../model/Interfaces";
+import { shoppingCartManipulation } from "./shopping";
+import { productEditManipulation } from "./edit";
+
+export interface AppState {
+  productListReducer: ProductListState;
```

Comment:
It is not the best that your state has properties with the name `blaReducer`. It should be like:

```
{
 products: ...
 productDetails: ...
 shoppingCart: ...
 editProduct: ...
}

```